papers/yellow/economy.tex

```diff
@@ -3,38 +3,37 @@ \section{Economic survivability}
-    \item{} Nothing should be kept in the state forever and for free.
-    Otherwise, the size of the state is always increasing, thus reducing clients performance.
+    \item{} coins protected by keys being lost should be returned into circulation.
+    Otherwise, after the end of the emission period, amount of the coins
+    in the circulation will always decrease and eventually reach zero.
+    \item{} nothing should be kept in the state forever and for free.
+    Otherwise, the size of the state is always increasing with time, thus reducing clients performance.
 \end{itemize}
 
-To achieve this, we propose the following modifications to the consensus rules.
-Register $R2$ of the box will contain tuple $(tx\_id, out\_num,
-last\_height)$. $last\_height$ field is used to determine the block height
-at the moment of transaction generation. Transaction can only be put in the
-block of height $h$ if for every created box $R2.last\_height < h$.
+To achieve this, we propose the following storage fee consensus rules.
+Register $R3$ of a box contains tuple $(creation\_height, tx\_id || out\_num)$, where $creation\_height$ provided by a user
+is used to determine the block height at the moment of transaction generation.
+Transaction can only be put in the block of height $h$ if for every created box its $creation\_height \le h$.
 
 Once the subsidized period for the box ends (that is,
-$current\_block\_height \ge R2.last\_height + SP$), anyone (presumably a miner) can
-create the new box with the exactly the same content (including the guarding
-script) except the monetary value and $R2$ content. The monetary value is
-reduced by $K \cdot B$, where $B$ is the box size and $K$ is the storage cost of one byte.
-Thus, $K \cdot B$ coins are to be paid to the miner.
-If box value is less that $K \cdot B$, all box content including tokens goes to miner.
-$R2.tx\_id$ and $R2.out\_num$ content of created box should contain correct transaction id
-and index, while $R2.last\_height$ should be set to $R2.last\_height$ of spending box
-plus $SP$, to ensure that subsidized period starts over.
+$current\_block\_height \ge box.creation\_height + SP$, see $SP$ value below), anyone~(presumably, a miner) can
+create a new box $out$ with the exactly the same content (including the guarding
+script) except the monetary value and $R3$ content. The monetary value is to
+reduced by $K \cdot B \cdot (out.creation\_height - self.creation\_height)$ maximum, where $B$ is the spent box~($self$)
+size and $K$ is the storage cost of one byte. Thus, the difference is to be paid to the miner.
+If box value is less that the storage fee, all the box content including tokens could be spent by the miner.
+
+For efficient lookup~(to find a proper output regarding an expired input without an iteration over transaction outputs),
+we require a spending proof for an expired box to be just a context extension which contains only an index of an
```

Comment:
From this 2 sentences it is still not clear, what is variable. I suppose to replace to something like:
```
...context extension which contains only a variable with id $127$ containing an index of an output  which is trying to spend the box.
```